### PR TITLE
Add screener page and contact facility page to cheetah flow

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR:
- Creates a page for the received dose screener question
- Creates a page for displaying nearby registered facilities
- Updates flow to go to contact facilities page if already received a dose

To test:
- Make sure va_online_scheduling_project_cheetah flag is turned on
- Go to http://localhost:3001/health-care/schedule-view-va-appointments/appointments/new-project-cheetah-booking
- Answer Yes or No on the second page to view different routing

## Testing done
Local testing and unit testing

## Screenshots
Screen Shot 2021-02-08 at 11 57 00 AM
Screen Shot 2021-02-08 at 11 57 05 AM
Screen Shot 2021-02-08 at 11 57 12 AM

## Acceptance criteria
- [ ] Pages render correctly, routing is correct
- [ ] Ticket AC are met

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/project-cheetah/flow.js [hunk 1]
```diff
@@ -10,7 +10,20 @@ export default {
...
+    next(state) {
+      const formData = selectProjectCheetahFormData(state);
+      if (formData.hasReceivedDose) {
+        return 'contactFacilities';
+      }
+      return 'vaFacility';
+    },
+  },
+  contactFacilities: {
+    url: '/new-project-cheetah-booking/contact-facilities',
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/flow.js:26]: This page is an end to the flow, so no `next`.

File: src/applications/vaos/project-cheetah/redux/actions.js [hunk 2]
```diff
@@ -449,6 +455,41 @@ export function onCalendarChange(selectedDates, pageKey) {
   };
 }
 
+export function openContactFacilitiesPage() {
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/redux/actions.js:458]: This is similar to the `openFacilitiesPage` action creator, except that there's no filtering by type of care. We just want to get any facilities where a user is registered.

File: src/applications/vaos/project-cheetah/redux/reducer.js [hunk 3]
```diff
@@ -477,6 +482,46 @@ export default function projectCheetahReducer(state = initialState, action) {
...
+    case FORM_PAGE_CONTACT_FACILITIES_OPEN_SUCCEEDED: {
+      let facilities = action.facilities;
+      const address = action.address;
+      const hasResidentialCoordinates =
+        !!action.address?.latitude && !!action.address?.longitude;
+      const sortMethod = hasResidentialCoordinates
+        ? FACILITY_SORT_METHODS.distanceFromResidential
+        : FACILITY_SORT_METHODS.alphabetical;
+
+      if (hasResidentialCoordinates && facilities.length) {
```
Comment (jbalboni) [inline src/applications/vaos/project-cheetah/redux/reducer.js:494]: Even though we can't switch sort methods, the AC has the facilities sorted by residential address, which alphabetical sorting as a fallback.

File: src/applications/vaos/tests/new-appointment/components/TypeOfVisitPage.unit.spec.jsx [hunk 4]
```diff
@@ -34,59 +34,62 @@ describe('VAOS <TypeOfVisitPage> ', () => {
...
-
-it('should not submit empty form', async () => {
-  const store = createTestStore(initialState);
-  const screen = renderWithStoreAndRouter(<TypeOfVisitPage />, {
-    store,
-  });
-  expect(await screen.findByText(/Continue/i)).to.exist;
 
-  fireEvent.click(screen.getByText(/Continue/));
+  it('should not submit empty form', async () => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/TypeOfVisitPage.unit.spec.jsx:38]: I copied and pasted the screener page tests from this file, and realized that the describe block was wrapping all the tests. So I fixed that. These changes are just indentation adjustments.

**Review by LanceGundersen — APPROVED** [review]
The two facility links go to a blank page:
image
Contact_a_facility___Veterans_Affairs

**Comment by jbalboni** [comment]
@LanceGundersen Those errors are because of facility locator issues when running locally, I believe. The same url used on staging should work.

**Pull request merged**